### PR TITLE
Update and extend beginner guide

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR updates the beginner guide to match the changes in #58 and extends it with another short description of the difference between `lookup()`, `insert()`, `ensure()` and `scdensure()`. While they are already somewhat described using comments in the code example, previous experience have shown that additional documentation is needed to ensure that new users of pygrametl learns the difference between these important methods. A link to the full API for all classes in `tables.py` has also been added so the methods documentation becomes easier to access.

### CODE REVIEW

File: docs/quickstart/beginner.rst [hunk 1]
```diff
@@ -251,14 +250,26 @@ of an ETL flow.
...
-data needed is included in the sales records. To ensure that the data is
-committed to the database and that the connection is closed correctly, the
-methods :meth:`.ConnectionWrapper.commit` and :meth:`.ConnectionWrapper.close`
-are executed at the end.
+Finally, the data can be inserted into the data warehouse. With a few
+exceptions, the dimension classes in pygrametl provide the same interface,
+likewise for the fact table classes. When adding members to dimensions or facts
+to fact tables in pygrametl, the methods :meth:`lookup`, :meth:`insert`,
+:meth:`ensure` are generally used. :meth:`lookup` returns the key for the member
+or fact that matches a given row if it exists, :meth:`insert` adds a member to a
```
Comment (chrthomsen) [inline docs/quickstart/beginner.rst:258]: on a `FactTable`, `lookup` returns both the keys and measure values

File: docs/quickstart/beginner.rst [hunk 2]
```diff
@@ -251,14 +250,26 @@ of an ETL flow.
...
+dimension or a fact to a fact table, and :meth:`ensure` performs a
+:meth:`lookup` followed by an :meth:`insert` if the member or fact does not
```
Comment (chrthomsen) [inline docs/quickstart/beginner.rst:260]: would it be easier to understand if "if the member or fact does not exist" was replaced with "if the `lookup` did not find an existing member or fact" or something similar to highlight that this depends on the outcome of `lookup`?

File: docs/quickstart/beginner.rst (hunk 1 above)
Comment (skejserjensen) [reply docs/quickstart/beginner.rst:258]: I think the sentence becomes much harder to read if we add a comma that is unrelated to the enumeration, I have tried to make it clear that difference things are returned by `lookup()` for members and facts using an `or`.

File: docs/quickstart/beginner.rst (hunk 2 above)
Comment (skejserjensen) [reply docs/quickstart/beginner.rst:260]: Good idea.

File: docs/quickstart/beginner.rst (hunk 1 above)
Comment (chrthomsen) [reply docs/quickstart/beginner.rst]: yes, I agree

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**